### PR TITLE
Integration of HTTP Client to provide easy access to the base API

### PR DESCRIPTION
**Pull request by ChristopherJamesMorris**
Search page components
Search page integration to use the API

### CODE REVIEW

File: modules/api/src/Service/TransparencySiteService.cs [hunk 1]
```diff
@@ -9,61 +9,62 @@
...
-            var parameters = BuildBlankSearchParameters();
+      parameters.Add("_site_definition_status_min", SiteStatus.LIVE, DbType.Int16, ParameterDirection.Input);
+      parameters.Add("_site_definition_status_max", SiteStatus.LIVE, DbType.Int16, ParameterDirection.Input);
 
-            parameters.Add("_site_definition_status_min", SiteStatus.LIVE, DbType.Int16, ParameterDirection.Input);
-            parameters.Add("_site_definition_status_max", SiteStatus.LIVE, DbType.Int16, ParameterDirection.Input);
+      if (!string.IsNullOrWhiteSpace(request.ProviderCode))
+      {
+        _logger.LogDebug("Preparing Query for ODS based search, {}", request.ProviderCode);
+        parameters.Add("_site_ods_code", request.ProviderCode.SearchAndReplace(new Dictionary<string, string> { { ",", "|" }, { " ", "|" }, { "| ", "|" } }).Replace("||", "|"), DbType.String, ParameterDirection.Input);
```
Comment (leew) [inline modules/api/src/Service/TransparencySiteService.cs:39]: It is not clear to me why this transformation is happening. 

Would it be better to have a named function in this class TransformInputToMultipleDBQueryStrings() or maybe this transformation should happen in the controller to a List<string> then the service here can handle the transformation to DB specific, it feels very hidden functionality at the minute. 

We would definitely need to have unit tests on this transformation

File: modules/api/src/Service/TransparencySiteService.cs [hunk 2]
```diff
@@ -9,61 +9,62 @@
...
+        parameters.Add("_site_name", null, DbType.String, ParameterDirection.Input);
+      }
+      else
+      {
+        _logger.LogDebug("Preparing Query for name based search, {}", request.ProviderName);
+        parameters.Add("_site_name", request.ProviderName.SearchAndReplace(new Dictionary<string, string> { { ",", "|" }, { "| ", "|" } }).Replace("||", "|"), DbType.String, ParameterDirection.Input);
```
Comment (leew) [inline modules/api/src/Service/TransparencySiteService.cs:45]: Duplicate of ln. 39

File: modules/api/src/Service/TransparencySiteService.cs (hunk 1 above)
Comment (leew) [reply modules/api/src/Service/TransparencySiteService.cs:39]: The more I think on this I would prefer the Controller to transform to a List of strings then the Service can concatenate back to a single DB parameter.

File: modules/end-user-portal/gpconnect-user-portal.Helpers/BooleanExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+using Microsoft.AspNetCore.Html;
```
Comment (leew) [inline modules/end-user-portal/gpconnect-user-portal.Helpers/BooleanExtensions.cs]: Move this Folder to within the src folder please, we are trying to remove interdependencies on each project.

File: modules/end-user-portal/src/Core/ApplicationBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+using Microsoft.AspNetCore.HttpOverrides;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Net.Http.Headers;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Core;
+
+public static class ApplicationBuilderExtensions
+{
+  public static void ConfigureApplicationBuilderServices(this IApplicationBuilder app, IWebHostEnvironment env)
```
Comment (leew) [inline modules/end-user-portal/src/Core/ApplicationBuilderExtensions.cs]: env is not used within this method, it can be removed

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    {
+      options.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/fhir+json"));
+      options.DefaultRequestHeaders.CacheControl = new CacheControlHeaderValue { NoCache = true };
+      options.BaseAddress = ValidBaseAddress(configuration["API_BASEADDRESS"]);
+    }).AddPolicyHandler(GetRetryPolicy())
+    .SetHandlerLifetime(TimeSpan.FromMinutes(5))
+    .ConfigurePrimaryHttpMessageHandler(() => CreateHttpMessageHandler(env));
+  }
+
+  private static IAsyncPolicy<HttpResponseMessage> GetRetryPolicy()
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Nice !

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  public static void AddHttpClientServices(IServiceCollection services, IConfiguration configuration, IWebHostEnvironment env)
+  {
+    services.AddHttpClient<IRequestService, RequestService>(options =>
+    {
+      options.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/fhir+json"));
+      options.DefaultRequestHeaders.CacheControl = new CacheControlHeaderValue { NoCache = true };
+      options.BaseAddress = ValidBaseAddress(configuration["API_BASEADDRESS"]);
+    }).AddPolicyHandler(GetRetryPolicy())
+    .SetHandlerLifetime(TimeSpan.FromMinutes(5))
+    .ConfigurePrimaryHttpMessageHandler(() => CreateHttpMessageHandler(env));
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: This can be done with a generic class type to remove the need for the additional method.

File: modules/end-user-portal/src/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+    }
+  },
+  "AllowedHosts": "*",
+  "ApplicationParameters": {
+    "PRODUCT_NAME": "GP Connect National Data Sharing Portal",
```
Comment (leew) [inline modules/end-user-portal/src/appsettings.json]: Config to follow JSON Pascal case please

File: modules/end-user-portal/src/wwwroot/css/site.css [hunk 8]
```diff
@@ -0,0 +1,9128 @@
+﻿@charset "UTF-8";
```
Comment (leew) [inline modules/end-user-portal/src/wwwroot/css/site.css:1]: Does this file need to be committed as it is generated from the scss file?

File: modules/end-user-portal/src/Pages/_ViewImports.cshtml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+﻿@using gpconnect_user_portal
```
Comment (leew) [inline modules/end-user-portal/src/Pages/_ViewImports.cshtml]: Mixed Namespaces

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+  private readonly JsonSerializerSettings _options;
+
+  public RequestService(ILogger<RequestService> logger, HttpClient httpClient)
+  {
+    _logger = logger;
+    _httpClient = httpClient;
+    _options = new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore };
+  }
+
+  public async Task<TResult> ExecuteApiGetAsync<TResult>(string url) where TResult : class
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: Can we have named functions for the HTTP calls please, it allows for easier unit testing. e.g. SearchSites, or whatever the function does. 

Having ExecuteApiGetAsync in front end code feels like a blurring of the seperation of concerns our UI code shouldn't care about the method of fetching the data

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs [hunk 11]
```diff
@@ -0,0 +1,95 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpRequestHandler.Interfaces;
+using Microsoft.AspNetCore.WebUtilities;
+using Newtonsoft.Json;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpRequestHandler;
+
+public class RequestService : IRequestService
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: I'm not sure on the name of this class, for me a class that calls a HTTP client should reference the API that it is calling. For example if calling Facebook it would be FacebookService or FacebookClient. 

I understand that this is not easy as we haven't really named our API...

File: modules/end-user-portal/README.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+# API
```
Comment (leew) [inline modules/end-user-portal/README.md]: This needs updating

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+using Polly;
+using Polly.Extensions.Http;
+using System.Net.Http.Headers;
+using System.Security.Authentication;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpRequestHandler;
+
+public static class HttpClientExtensions
+{
+  public static void AddHttpClientServices(IServiceCollection services, IConfiguration configuration, IWebHostEnvironment env)
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Can we make this a true extension method on IServiceCollection please

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  private static HttpMessageHandler CreateHttpMessageHandler(IWebHostEnvironment env)
+  {
+    var httpClientHandler = new HttpClientHandler();
+    httpClientHandler.SslProtocols = SslProtocols.Tls13 | SslProtocols.Tls12 | SslProtocols.Tls11 | SslProtocols.Tls;
+    return httpClientHandler;
+  }
+
+  private static Uri ValidBaseAddress(string baseAddress)
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: This can be replaced with https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder.-ctor?view=net-6.0#system-uribuilder-ctor(system-string)

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+  public async Task<TResult> ExecuteApiGetAsync<TRequest, TResult>(TRequest t, string url) where TRequest : class where TResult : class
+  {
+    var cancellationTokenSource = new CancellationTokenSource();
+    var request = new HttpRequestMessage()
+    {
+      Method = HttpMethod.Get
+    };
+    var result = default(TResult);
+
+    var jsonRequest = JsonConvert.SerializeObject(t, _options);
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: ln 60 -> 65 could go to a helper method, ConvertObjToQueryParams, although it's only going to be used a couple of times so maybe not worth it.

Just need to make sure that we check the URI contains the correct params.

File: modules/end-user-portal/src/Core/WebConfigurationBuilder.cs [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+{
+  public static void ConfigureWebHostDefaults(IWebHostBuilder webHostDefaultsBuilder)
+  {
+    webHostDefaultsBuilder.UseKestrel(options =>
+        {
+          options.AddServerHeader = false;
+        });
+  }
+
+  public static void ConfigureWebHost(IWebHostBuilder webHostBuilder)
```
Comment (leew) [inline modules/end-user-portal/src/Core/WebConfigurationBuilder.cs]: ConfigureErrorHandling

File: modules/end-user-portal/src/Core/WebConfigurationBuilder.cs [hunk 17]
```diff
@@ -0,0 +1,17 @@
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Core;
+
+public static class WebConfigurationBuilder
+{
+  public static void ConfigureWebHostDefaults(IWebHostBuilder webHostDefaultsBuilder)
```
Comment (leew) [inline modules/end-user-portal/src/Core/WebConfigurationBuilder.cs]: ConfigureKestrel?

File: modules/end-user-portal/src/Pages/Header.cshtml.cs [hunk 18]
```diff
@@ -0,0 +1,20 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models;
+using Microsoft.Extensions.Options;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Pages
+{
+  public class HeaderModel : BaseModel
+    {
+        private readonly ILogger<HeaderModel> _logger;
```
Comment (leew) [inline modules/end-user-portal/src/Pages/Header.cshtml.cs]: Do we need the logger in this class as it is not used?

File: modules/end-user-portal/src/Pages/Navigation.cshtml.cs [hunk 19]
```diff
@@ -0,0 +1,20 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models;
+using Microsoft.Extensions.Options;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Pages
+{
+  public class NavigationModel : BaseModel
+    {
+        private readonly ILogger<NavigationModel> _logger;
```
Comment (leew) [inline modules/end-user-portal/src/Pages/Navigation.cshtml.cs]: Do we need the logger on this page now?

File: modules/api/src/Service/TransparencySiteService.cs [hunk 20]
```diff
@@ -9,61 +9,62 @@
 using System.Collections.Generic;
```
Comment (leew) [inline modules/api/src/Service/TransparencySiteService.cs:9]: These changes could be in a separate pull request. With the focus for this one on the UI.

File: modules/end-user-portal/src/Pages/Header.cshtml [hunk 21]
```diff
@@ -0,0 +1,20 @@
+<header class="nhsuk-header" role="banner">
+    <partial name="_CookieConsentPartial" />
+    <div class="nhsuk-width-container nhsuk-header__container">
+        <div class="nhsuk-header__logo">
+            <a class="nhsuk-header__link nhsuk-header__link--service" href="/" aria-label="@Model.ProductName">
+                <svg class="nhsuk-logo" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 16" height="40" width="100">
```
Comment (leew) [inline modules/end-user-portal/src/Pages/Header.cshtml:6]: The nhs logo is an SVG, I would have thought that this would have been controlled centrally rather than implemented in every project.

File: modules/end-user-portal/src/Pages/Navigation.cshtml [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+            <li class="nhsuk-header__navigation-item">
+                <a class="nhsuk-header__navigation-link" href="/">
+                    <text>Home</text>
+                    <svg class="nhsuk-icon nhsuk-icon__chevron-right" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" width="34" height="34">
+                        <path d="M15.5 12a1 1 0 0 1-.29.71l-5 5a1 1 0 0 1-1.42-1.42l4.3-4.29-4.3-4.29a1 1 0 0 1 1.42-1.42l5 5a1 1 0 0 1 .29.71z"></path>
+                    </svg>
+                </a>
+            </li>
+            <li class="nhsuk-header__navigation-item">
+                <a class="nhsuk-header__navigation-link" href="/Help">
```
Comment (leew) [inline modules/end-user-portal/src/Pages/Navigation.cshtml]: We don't currently have a help page?

File: modules/end-user-portal/src/Pages/Search/Index.cshtml.cs [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpRequestHandler.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Pages;
+
+public partial class SearchModel : BaseModel
+{
+  private readonly ILogger<SearchModel> _logger;
```
Comment (leew) [inline modules/end-user-portal/src/Pages/Search/Index.cshtml.cs]: We are not logging anything out of this class, if there is nothing to log can we remove the logger please.

File: modules/end-user-portal/src/Pages/Search/Index.cshtml.cs [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+  public async Task<IActionResult> OnGet()
+  {
+    ClearModelState();
+    await PopulateControls();
+    return Page();
+  }
+
+  private async Task PopulateControls()
+  {
+    CcgList = await GetCCGs();
```
Comment (leew) [inline modules/end-user-portal/src/Pages/Search/Index.cshtml.cs]: Do we still need this request, this is done on every Get and Search (Post) Request to the page, for what is very likely to be 'static' data.

At least we are only planning on updating this server side once every 2 days

**Review by leew — CHANGES_REQUESTED** [review]

File: modules/api/src/Service/TransparencySiteService.cs (hunk 1 above)
Comment (ChristopherJamesMorris) [reply modules/api/src/Service/TransparencySiteService.cs:39]: I've created a new ticket GCNDSP-272 to address these changes.

File: modules/api/src/Service/TransparencySiteService.cs (hunk 2 above)
Comment (ChristopherJamesMorris) [reply modules/api/src/Service/TransparencySiteService.cs:45]: As above.

File: modules/end-user-portal/src/Core/ApplicationBuilderExtensions.cs (hunk 4 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/ApplicationBuilderExtensions.cs]: Sorted.

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs (hunk 5 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Thanks :-)

File: modules/end-user-portal/src/Pages/Navigation.cshtml (hunk 22 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/Navigation.cshtml]: Removed.

File: modules/end-user-portal/src/Pages/Search/Index.cshtml.cs (hunk 23 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/Search/Index.cshtml.cs]: Removed.

File: modules/end-user-portal/src/Pages/Search/Index.cshtml.cs (hunk 24 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/Search/Index.cshtml.cs]: Removed for the time being. It may be required in the future but for now I've removed all references to CCGs

File: modules/end-user-portal/src/Pages/Header.cshtml (hunk 21 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/Header.cshtml:6]: It doesn't appear so - I've checked the design manual and it just references the markup as shown here.

File: modules/api/src/Service/TransparencySiteService.cs (hunk 20 above)
Comment (ChristopherJamesMorris) [reply modules/api/src/Service/TransparencySiteService.cs:9]: True - but we can cover them off as part of the other ticket for the API refinements.

File: modules/end-user-portal/gpconnect-user-portal.Helpers/BooleanExtensions.cs (hunk 3 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/gpconnect-user-portal.Helpers/BooleanExtensions.cs]: Done.

File: modules/end-user-portal/src/Pages/Header.cshtml (hunk 21 above)
Comment (leew) [reply modules/end-user-portal/src/Pages/Header.cshtml:6]: Thank you for checking

File: modules/end-user-portal/src/Pages/Search/Index.cshtml.cs (hunk 24 above)
Comment (leew) [reply modules/end-user-portal/src/Pages/Search/Index.cshtml.cs]: Thank you, I completely agree.

File: modules/end-user-portal/gpconnect-user-portal.Helpers/BooleanExtensions.cs (hunk 3 above)
Comment (leew) [reply modules/end-user-portal/gpconnect-user-portal.Helpers/BooleanExtensions.cs]: Thank you

File: modules/api/src/Service/TransparencySiteService.cs (hunk 20 above)
Comment (leew) [reply modules/api/src/Service/TransparencySiteService.cs:9]: I think that we agree that these changes are not going in this commit?

File: modules/end-user-portal/src/appsettings.json (hunk 7 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/appsettings.json]: Sorted.

File: modules/end-user-portal/src/Pages/_ViewImports.cshtml (hunk 9 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/_ViewImports.cshtml]: Sorted.

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs (hunk 15 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: I've put this into an helper extension method.

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs (hunk 10 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: Changed as per discussion

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs (hunk 11 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: Changed as per discussion

File: modules/end-user-portal/src/Pages/Navigation.cshtml.cs (hunk 19 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/Navigation.cshtml.cs]: This has been removed.

File: modules/end-user-portal/src/Pages/Header.cshtml.cs (hunk 18 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Pages/Header.cshtml.cs]: Removed

File: modules/end-user-portal/src/Core/WebConfigurationBuilder.cs (hunk 17 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/WebConfigurationBuilder.cs]: Removed.

File: modules/end-user-portal/src/Core/WebConfigurationBuilder.cs (hunk 16 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/WebConfigurationBuilder.cs]: I've removed this now.

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs (hunk 14 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Done.

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs (hunk 13 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Done.

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs (hunk 6 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: I've amended this.

File: modules/end-user-portal/src/Core/HttpClientServices/SiteService.cs [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+  private readonly JsonSerializerSettings _options;
+
+  public SiteService(ILogger<SiteService> logger, HttpClient httpClient)
+  {
+    _logger = logger;
+    _httpClient = httpClient;
+    _options = new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore };
+  }
+
+  public async Task<TResult> SearchSitesAsync<TRequest, TResult>(TRequest searchRequest) where TRequest : class where TResult : class
```
Comment (leew) [inline modules/end-user-portal/src/Core/HttpClientServices/SiteService.cs]: This doesn't need to be generic any more we know that we will send a SearchRequest and get back a SiteResults object

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs (hunk 6 above)
Comment (leew) [reply modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Thanks

File: modules/end-user-portal/src/appsettings.json (hunk 7 above)
Comment (leew) [reply modules/end-user-portal/src/appsettings.json]: thanks

File: modules/end-user-portal/README.md (hunk 12 above)
Comment (leew) [reply modules/end-user-portal/README.md]: thank you

File: modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs (hunk 15 above)
Comment (leew) [reply modules/end-user-portal/src/Core/HttpRequestHandler/RequestService.cs]: Perfect thank you

File: modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs (hunk 14 above)
Comment (leew) [reply modules/end-user-portal/src/Core/HttpRequestHandler/HttpClientExtensions.cs]: Sorry Chris, I was meaning that the method ValidBaseAddress Method could be replaced with the UriBuilder rather than the contents of the method. 

But this is minor.

File: modules/end-user-portal/src/Core/HttpClientServices/SiteService.cs (hunk 25 above)
Comment (ChristopherJamesMorris) [reply modules/end-user-portal/src/Core/HttpClientServices/SiteService.cs]: Changed to be non generic
